hooks/publish.py

```diff
@@ -976,18 +1006,23 @@ def _get_publish_linked_entity_name(self, task_settings, item):
             pub_linked_entity_name_tmpl = publisher.get_template_by_name(publish_linked_entity_name_template)
             if not pub_linked_entity_name_tmpl:
                 # this template was not found in the template config!
-                raise TankError("The Template '%s' does not exist!" % publish_linked_entity_name_template)
+                raise TankMissingTemplateError("The Template '%s' does not exist" % publish_linked_entity_name_template)
 
             # First get the fields from the context
             try:
-                fields.update(item.context.as_template_fields(pub_linked_entity_name_tmpl))
+                context_fields = item.context.as_template_fields(pub_linked_entity_name_tmpl)
+                # store the fields resolved using the context for this template
+                item.properties.publish_linked_entity_name_context_fields = context_fields
```

Comment:
`context_fields` is the same regardless of what template you are processing. You can just calculate this once where you need it rather than overloading the item properties to store it.